crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -300,10 +293,50 @@ impl Persistence {
             id: auction_id,
             orders,
             prices,
-            deadline,
             surplus_capturing_jit_order_owners,
         })
     }
+
+    /// Get competition winner.
+    pub async fn get_winner(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<(eth::Address, competition::Score, eth::BlockNo), error::Winner> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_winner"])
+            .start_timer();
+
+        let mut ex = self
+            .postgres
+            .pool
+            .begin()
+            .await
+            .context("begin")
+            .map_err(error::Winner::DbError)?;
+
+        let (winner, score, deadline) = {
+            let scores = database::settlement_scores::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch scores")
+                .map_err(error::Winner::DbError)?
+                .ok_or(error::Winner::Missing)?;
+            (
+                H160(scores.winner.0).into(),
+                competition::Score::new(
+                    big_decimal_to_u256(&scores.winning_score)
+                        .ok_or(error::Winner::InvalidScore(anyhow::anyhow!(
+                            "database score"
+                        )))?
+                        .into(),
+                )
+                .map_err(|_| error::Winner::InvalidScore(anyhow::anyhow!("zero score")))?,
```

Comment:
nit: extract into a variable since it is quite hard to read between the tuple elements.

Reply:
fixed